.github/workflows/codeql.yml

```
@@ -0,0 +1,41 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+  schedule:
+    - cron: '43 0 * * 6'
```

same comment here, `cron` isn't necessary imo and we can do it on any PR

```suggestion
on:
  push:
    branches: [ "master" ]
  pull_request:
```